Ip/Internal/Design/Model.php

```diff
@@ -312,7 +312,14 @@ public static function getThemeLayouts()
         foreach ($files as $filename) {
             if ('php' == strtolower(pathinfo($filename, PATHINFO_EXTENSION))) {
                 if ($filename[0] != '_') {
-                    $layouts[] = $filename;
+                    $file_contents = file_get_contents($themeDir.$filename);
+                    preg_match_all("(@Layout Name:(.*)\n)siU", $file_contents, $file_layout);
+                    if (isset($file_layout[1]) && isset($file_layout[1][0]) && !empty($file_layout[1][0])) {
+                        $layout_name = preg_replace('/[^a-zA-Z0-9\s]/', '', $file_layout[1][0]);
```

Comment:
I believe this preg_replace is redundant. I could imagine some russian agency building a custom template for their clients and willing to name the layout in cyrillic. In that case, layout name will become empty. I see no security or other reasons to restrict any characters in layout title. Would you mind if I remove this line?